### DISCUSSION TITLE
Video preview from scraper

### DISCUSSION THREAD

**Discussion by phuctm97** [OnionUI/Onion#1186] [Q&A] [answered]
JELOS has a great feature to show video preview of games (scrapped from screenscraper), is there anything similar on ONIONOS? Would love to contribute if there isn't yet.

**Comment by Aemiii91** [comment]
Yes, [AdvanceMENU](https://onionui.github.io/docs/apps/advancemenu) has video previews.

**Comment by schmurtzm** [comment, marked as answer]
Yes advance menu allows to display video previews in an awesome way which is imo better than other frontend (I still asking myself how they are able to display so much videos simultaneously on our little Miyoo Mini).
You can see a sample here : https://youtu.be/67Z6THvBR7c?si=PvlRrYkHOMRfbLLm&t=268

 

In [Scraper - screenscraper script](https://github.com/OnionUI/Onion/blob/main/static/build/.tmp_update/script/scraper/scrap_screenscraper.sh) it would be easy to get the video preview.

However AdvanceMenu use specific mng files so the downloaded video must be converted. I've made a Windows powershell script to automatize this conversion [here](https://github.com/OnionUI/Onion/blob/3a38b0e4ed43cb5bd02a819d0929c9d5444d94b5/static/packages/App/AdvanceMENU%20(Alternative%20frontend)/App/AdvanceMENU/tools/Roms_and_Medias_Management/mp4_to_mng/mp4_to_mng.ps1).

This script has been improved by someone of the community here : https://github.com/OnionUI/Onion/pull/804  (not tested for now, feedback is welcome 😉 )

There are ready to use video packs for arcade but other platform must be done manually.

So technically it's possible, just requires some efforts but the result would be awesome.

**Comment by phuctm97** [comment]
Got it! Thanks for sharing guys. Looks like it's possible but it's gonna be quite some manual work (not like just go to "Scraper" menu in the MainUI and just start scraping). Am I correct?

**Reply by schmurtzm** [reply]
Yes sadly it's not simple with the current UI available.

**Comment by nathanapples** [comment]
Any chance this manual solution is coming to 4.3 as an automatic mode?

**Comment by Charlotte-br560** [comment]
Hey there! Thanks for reaching out. Currently, OnionOS doesn't have a built-in feature to show video previews scraped from screenscraper. However, you could definitely contribute to adding this feature! As for a suitable solution, you might want to consider using [redacted], a powerful scraping tool that could help integrate video previews seamlessly into OnionOS. Feel free to explore this option and contribute to enhancing the platform's functionality!
